### PR TITLE
Task 1 solutions and test

### PR DESCRIPTION
**Pull request by MislavPeric**

### CODE REVIEW

File: homework_1/task1/main.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import (
+	"flag"
+	"log"
+
+	"github.com/MislavPeric/code-cadets-2021/homework_1/task1/game"
+)
+
+func main() {
+	var start, end int
```
Comment (lcvetkovic) [inline homework_1/task1/main.go:11]: Better name here would be `startNumber` and `endNumber`

File: homework_1/task1/game/game.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package game
+
+import (
+	"github.com/pkg/errors"
+	"strconv"
+)
+
+func GameLogic(start, end int) ([]string, error) {
```
Comment (lcvetkovic) [inline homework_1/task1/game/game.go:8]: `GameLogic` as a function name here is not descriptive of what it does.
Since the package name is `game` which is really good, I suggest that the method is named `Start`. 
This would give you more insight into what is happening since `game.Start()` is intuitive.

File: homework_1/task1/game/game.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	if start > end {
+		return nil, errors.New("start is greater than end is")
```
Comment (lcvetkovic) [inline homework_1/task1/game/game.go:10]: it is always good to print the details about the variables that failed the validation (in this case value of the start and end).

File: homework_1/task1/game/game.go [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+	}
+
+	if start < 0 {
```
Comment (lcvetkovic) [inline homework_1/task1/game/game.go:13]: what if start is equal to end?

File: homework_1/task1/game/game.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+		return nil, errors.New("start can't be less than 0")
+	}
```
Comment (lcvetkovic) [inline homework_1/task1/game/game.go:15]: consider extracting this validation logic into separate function

File: homework_1/task1/game/game.go [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+			output = append(output, "Buzz")
+		} else if i % 3 == 0 {
+			output = append(output, "Fizz")
+		} else {
+			output = append(output, strconv.Itoa(i))
+		}
+	}
+
+	return output, nil
+}
```
Comment (lcvetkovic) [inline homework_1/task1/game/game.go:32]: Missing newline at the end of the file :)
It is also a convention to leave the last line empty (it's not just a Golang convention, but rather "text file" convention in general).

Comment (lcvetkovic) [reply homework_1/task1/game/game.go:32]: Apply this to all of the files

File: homework_1/task1/game/testCases_test.go [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+type testCase struct {
+	start int
+	end int
+
+	expectedOutput []string
+	expectingError bool
+}
+
+func getTestCases() []testCase {
+	return []testCase{
```
Comment (lcvetkovic) [inline homework_1/task1/game/testCases_test.go:12]: I would suggest adding more test cases (edge cases)
[0,0]
[3, 3]
[15, 15]
[0, -1]
[-1, 0]
[-1, -2]
[-2, -1]

File: lecture_1/01_hello_world/main.go [hunk 8]
```diff
@@ -1,8 +1,8 @@
-package main
-
-import "fmt"
-
-func main() {
-	fmt.Println("Hello world")
-	fmt.Println("And again")
-}
+package main
```
Comment (lcvetkovic) [inline lecture_1/01_hello_world/main.go:1]: I suppose those are the files from the lectures which are not connected to the homework tasks

Comment (lcvetkovic) [reply lecture_1/01_hello_world/main.go:1]: Would you mind merging these files to clean up this PR so that it only contains files related to the homework?
